### DISCUSSION TITLE
Raster drawing and color manipulation

### DISCUSSION THREAD

**Discussion by JosephTRichardson** [pelicanmapping/osgearth#2172] [Support Q&A] [answered]
Hi. I'm, apologetically, very much a noob to osgEarth, and I wanted to ask for some general pointers.

I'm maintainer of an application (which I didn't write) that reads elevation rasters (DTED or other GDAL-compatible) and draws them on a 2D plane (using Qt and QPainter). It renders the elevation range using a color gradient, and features tools to edit the color table (e.g. change the colors at various points along the gradient), and re-render the gradient of a selected local area using the local min and max. 

I've been tasked with adapting this 2D application as a plugin that does much the same thing on a 3D globe for an osgEarth-based application. And I feel pretty out to sea, unsure even where to start. Are there any osgEarth functions that might help me draw rasters or manipulate their color rendering? Or, at the very least, where would I begin with drawing rasters on the globe using WGS 84 coordinates?

**Comment by emminizer** [comment, marked as answer]
Hello! Is this for SIMDIS?

osgEarth lets you load GDAL-compatible layers directly in the .earth file. Once you have elevation data loaded in the .earth file, SIMDIS does provide the ability to do hill shade and color reliefs from the Layer Manager. Expand "Elevation" then right-click your layer and choose "Add Color Relief". You can edit the gradients that are being used in the Gradient Manager (Tools / Gradients).

You can also add hill shading using the Layer Manager.

You can also do this your own in your .earth file. That's what SIMDIS is doing under the hood. See https://github.com/gwaldron/osgearth/blob/master/tests/gdaldem.earth as an example.

I can't easily answer the DTED question, as I am not personally familiar with using DTED directly. Everything I use has been processed into MBTiles format by the time I get it for elevation data.

Hope this helps.

 - Dan

**Reply by JosephTRichardson** [reply]
Hi Dan, thanks for the quick reply. It's not SIMDIS exactly, but it does use osgEarth through the SIMDIS SDK, so that's the right ballpark. It means, though, that we don't have access to the SIMDIS Layer Manager, but ought to be able to do what SIMDIS does. It does use .earth files, so thanks for the tip. I will check out those .earth file options.
